core/client/src/call_executor.rs

```diff
@@ -139,6 +175,30 @@ where
 		Ok(CallResult { return_data, changes })
 	}
 
+	fn contextual_call<
+		PB: Fn() -> error::Result<Block::Header>,
+		EM: Fn(Result<Vec<u8>, Self::Error>, Result<Vec<u8>, Self::Error>) -> Result<Vec<u8>, Self::Error>,
+	>(
+		&self,
+		at: &BlockId<Block>,
+		method: &str,
+		call_data: &[u8],
+		changes: &mut OverlayedChanges,
+		initialised_block: &mut Option<BlockId<Block>>,
+		prepare_environment_block: PB,
+		manager: ExecutionManager<EM>,
+	) -> Result<Vec<u8>, error::Error> where ExecutionManager<EM>: Clone {
+		let state = self.backend.state_at(*at)?;
+		//TODO: Find a better way to prevent double block initialization
```

Comment:
This isn't a new TODO (moved from `client.rs`). Could add a tracking issue, if required, but probably better for TODO author to do this.